src/components/RecipeEdit.vue

```diff
@@ -373,7 +390,39 @@ export default {
         this.$root.$on('reloadRecipeEdit', () => {
             this.loadRecipeData()
         })
+        // Register recipe-reference selection hook for showing popup when requested
+        // from a child element
+        this.$off('showRecipeReferencesPopup')
+        this.$on('showRecipeReferencesPopup', (val) => {
+            this.referencesPopupFocused = true
+            this.popupContext = val
+        })
+        // Register hook when recipe reference has been selected in popup
+        this.$off('ms-popup-selected')
+        this.$on('ms-popup-selected', (opt) => {
+            this.referencesPopupFocused = false
+            this.popupContext.context.pasteString('r/' + opt.recipe_id + ' ')
+        })
+        // Register hook when recipe reference has been selected in popup
+        this.$off('ms-popup-selection-canceled')
+        this.$on('ms-popup-selection-canceled', (opt) => {
+            this.referencesPopupFocused = false
+            this.popupContext.context.pasteCanceled()
+        })
         this.savingRecipe = false
+        
+        // Load data for all recipes to be used in recipe-reference popup suggestions
+        axios.get(this.$window.baseUrl + '/api/recipes')
+            .then(function (response) {
+                $this.allRecipes = response.data
+            })
+            .catch(function (e) {
+                console.log(e)
+            })
+            .then(() => {
+                // finally
+                $this.loadingRecipeReferences = false
+            })
```

Comment:
This is something we have in the global Vuex state, don't we?

Reply:
We don’t, but I when implementing this I was wondering if we should. Then we wouldn’t have to fetch the data every time we open the editor. On the other hand, if we fetch it every time, we can be sure that the data is up to data (e.g. if someone else added a new recipe).

Reply:
Ahh, we have only the current recipe there. My bad, sorry. I would say we keep it like this. Doing more caching might be later an option if performance gets troublesome.